conan/tools/cmake/toolchain/blocks.py

```diff
@@ -328,8 +328,64 @@ class AppleSystemBlock(Block):
         # Setting CMAKE_OSX_DEPLOYMENT_TARGET if "os.version" is defined by the used conan profile
         set(CMAKE_OSX_DEPLOYMENT_TARGET "{{ CMAKE_OSX_DEPLOYMENT_TARGET }}" CACHE STRING "")
         {% endif %}
+        {% if enable_bitcode is defined and enable_bitcode is sameas true %}
+        # Bitcode
+        set(CMAKE_XCODE_ATTRIBUTE_ENABLE_BITCODE YES)
+        set(CMAKE_XCODE_ATTRIBUTE_BITCODE_GENERATION_MODE bitcode)
+        {% endif %}
+        {% if enable_arc is defined and enable_arc is sameas true %}
+        # ARC
+        set(CMAKE_XCODE_ATTRIBUTE_CLANG_ENABLE_OBJC_ARC YES)
+        {% elif enable_arc is defined and enable_arc is sameas false %}
+        # ARC
+        set(CMAKE_XCODE_ATTRIBUTE_CLANG_ENABLE_OBJC_ARC NO)
+        {% endif %}
+        {% if enable_visibility is defined and enable_visibility is sameas true %}
+        # Visibility
+        set(CMAKE_XCODE_ATTRIBUTE_GCC_SYMBOLS_PRIVATE_EXTERN YES)
+        {% elif enable_visibility is defined and enable_visibility is sameas false %}
+        # Visibility
+        set(CMAKE_XCODE_ATTRIBUTE_GCC_SYMBOLS_PRIVATE_EXTERN NO)
+        {% endif %}
+        {% if common_flags|length > 0 %}
+        string(APPEND CONAN_CXX_FLAGS "{% for flag in common_flags %} {{ flag }}{% endfor %}")
+        string(APPEND CONAN_C_FLAGS "{% for flag in common_flags %} {{ flag }}{% endfor %}")
+        string(APPEND CONAN_SHARED_LINKER_FLAGS "{% for flag in common_flags %} {{ flag }}{% endfor %}")
+        string(APPEND CONAN_EXE_LINKER_FLAGS "{% for flag in common_flags %} {{ flag }}{% endfor %}")
+        {% endif %}
         """)
 
+    def _get_xcode_common_flags(self):
+        """
+        Adding Xcode flags if enabled Bitcode, ARC or Visibility via conf.
+
+        :return: `dict` with common flags and boolean values if enabled bitcode, arc or visibility.
+        """
+        # Issue related: https://github.com/conan-io/conan/issues/9448
+        enable_bitcode = self._conanfile.conf.get("tools.apple:enable_bitcode",
+                                                  default=self._conanfile.settings.get_safe('os') in ["watchOS", "tvOS"],
+                                                  check_type=bool)
+        enable_arc = self._conanfile.conf.get("tools.apple:enable_arc", check_type=bool)
+        enable_visibility = self._conanfile.conf.get("tools.apple:enable_visibility", check_type=bool)
```

Comment:
I don't this one is apple specific at all

Reply:
I don't really know how to put it. You're right, it's not specific to Apple but we're only using it for this one for now. You can always inject any flags thanks to `conf` so to convert it into an enum variable perhaps it's not needed as well...

Reply:
I think it will introduce more confusion than needed. it would be better to do it right from the very beginning.
otherwise, users will ask, what will take more priority (`tools.apple:enable_visibility` vs flags injected in conf), and what happens if they conflict...

I would propose `tools.gnu:visiibility` with values `default|internal|hidden|protected`